src/ucp/proto/proto_am.inl

```diff
@@ -103,7 +109,20 @@ ucs_status_t ucp_do_am_bcopy_multi(uct_pending_req_t *self, uint8_t am_id_first,
                 return (ucs_status_t)packed_len;
             }
         } else {
-            return UCS_INPROGRESS;
+            ucs_assertv(/* The offset has to be less than the data length for
+                         * the first and middle segments */
+                        ((req->send.state.dt.offset != req->send.length) &&
```

Comment:
why not just `<`?

Reply:
can use just `req->send.state.dt.offset <= req->send.length`
looks like it does not check what you expected

Reply:
yes, fixed to just compare for `req->send.state.dt.offset <= req->send.length`